libcontainer/intelrdt/intelrdt.go

```diff
@@ -482,8 +492,9 @@ func (m *Manager) Apply(pid int) (err error) {
 func (m *Manager) Destroy() error {
 	// Don't remove resctrl group if closid has been explicitly specified. The
 	// group is likely externally managed, i.e. by some other entity than us.
-	// There are probably other containers/tasks sharing the same group.
-	if m.config.IntelRdt != nil && m.config.IntelRdt.ClosID == "" {
+	// There are probably other containers/tasks sharing the same group. Also
+	// only remove the directory if it was created by us.
+	if m.config.IntelRdt != nil && m.config.IntelRdt.ClosID == "" && m.directoryCreated {
 		m.mu.Lock()
 		defer m.mu.Unlock()
 		if err := os.RemoveAll(m.GetPath()); err != nil {
```

Comment:
not really about this PR, but this can benefit from `cgroups.RemovePath`, as here we only want to remove directories, not files.